base/broadcast.jl

```diff
@@ -279,10 +280,30 @@ end
     return C
 end
 
-eltypestuple(a) = (Base.@_pure_meta; Tuple{eltype(a)})
-eltypestuple(T::Type) = (Base.@_pure_meta; Tuple{Type{T}})
-eltypestuple(a, b...) = (Base.@_pure_meta; Tuple{eltypestuple(a).types..., eltypestuple(b...).types...})
-_broadcast_eltype(f, A, Bs...) = Base._return_type(f, eltypestuple(A, Bs...))
+maptuple(f) = Tuple{}
+maptuple(f, a, b...) = Tuple{f(a), maptuple(f, b...).types...}
```

Comment:
IIUC, this function returns a tuple type collecting the values returned when the passed function is applied to each of the trailing arguments of this function (but only where you can construct a tuple type including those values, e.g. not where the passed function returns strings). Perhaps a slightly more descriptive / narrower name would be better, e.g. `maptotupletype` or `maptoTuple`? (I imagine you can come up with something better than those suggestions.) Or do I misunderstand, and `maptuple` somehow better describes this function?